src/modules/machine/actions/components/LanguageSwitcher.tsx

```diff
@@ -0,0 +1,50 @@
+import { Box, Button } from "@mui/material";
+import { useEffect, useState } from "react";
+import { useAppDispatch, useAppSelector } from "../../../core/store/hooks";
+import {
+  Language,
+  languageChange,
+  languageSwitcherSelector,
+} from "../store/languageSwitcherSlice";
+
+const LanguageSwitcher = () => {
+  const [lang, setLang] = useState<Language>({ lang: "en" });
+  const [isSelectedEn, setIsSelectedEn] = useState<boolean>(true);

```

Comment:
What if you would have 10 languages. In this case you will need to create a useState for each of them.